android/tv-custom/app/src/main/java/com/ttd/sso/sample/ui/AuthScreenViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package com.ttd.sso.sample.ui
+
+import android.content.Context
+import android.util.Log
+import androidx.activity.result.ActivityResultRegistry
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.myopenpass.auth.OpenPassManager
+import com.myopenpass.auth.OpenPassManagerState
+import com.myopenpass.auth.tv.flow.TvSignInFlowClient
+import com.ttd.sso.sample.ui.AuthScreenState.ErrorState
+import com.ttd.sso.sample.ui.AuthScreenState.Loading
+import com.ttd.sso.sample.ui.AuthScreenState.SignedInState
+import com.ttd.sso.sample.ui.AuthScreenState.SignedOutState
+import kotlinx.coroutines.launch
+
+sealed interface AuthScreenState : ViewState {
+    data object Loading : AuthScreenState
+    data class SignedInState(val email: String) : AuthScreenState
+    data object SignedOutState : AuthScreenState
+    data class ErrorState(val error: Throwable) : AuthScreenState
+}
+
+class AuthScreenViewModel(
+    registry: ActivityResultRegistry,
+    owner: LifecycleOwner,
+    private val manager: OpenPassManager = OpenPassManager.getInstance(),
+) : BasicViewModel<AuthScreenState>(Loading) {
+
+    // We create a single instance of the TvSignInFlowClient which will be re-used across all attempts. This is
+    // required because the ActivityResultRegistry must be used *before* the LifecycleOwner has been created.
+    private val client: TvSignInFlowClient = TvSignInFlowClient(CLIENT_ID, manager, registry, owner)
```

Comment:
move this to be a dependency in the constructor, it'd be similar to what folks would do via their DI systems

Reply:
Done, thanks